### PR TITLE
[BACKLOG-20636] Have a CDE dashboard rendering service in PDI

### PR DESCRIPTION
**Pull request by afrjorge**
- fix ehcache initialization issues
  - updated ehcache to version 2.5.1
  - removed ehcache dependency from core

### CODE REVIEW

**Comment by afrjorge** [comment]
@graimundo @eliofreitas please review

File: osgi/impl/src/test/java/org/pentaho/ctools/cde/api/RenderApiTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      assertTrue( renderApi.getDashboard(
+        DASHBOARD_WCDF_PATH, inferScheme, root, absolute, bypassCache, debug, DEFAULT_SCHEME, style, alias, request  )
+          .startsWith( expected ) );
+    } catch( Exception e ) {
+      fail( "No exception should be thrown" );
+    }
+  }
+
+  @Test
+  public void testGetDashboardParameters() throws Exception {
```
Comment (davidmsantos90) [inline osgi/impl/src/test/java/org/pentaho/ctools/cde/api/RenderApiTest.java]: Same for this one

File: osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  }
+
+  @Test
+  public void testGetUserSession() {
+    assertNull( cdeEnvironment.getUserSession() );
+  }
+
+  @Test
+  public void testRefresh() {
+    cdeEnvironment.refresh();
```
Comment (davidmsantos90) [inline osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java]: Is this testing an error? If so, it should be explicit in the test declaration

File: osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  }
+
+  @Test
+  public void testGetRepositoryBaseContentUrl() {
+    final String expected = "/plugin/cde/res/";
+    assertEquals( expected, cdeEnvironment.getRepositoryBaseContentUrl() );
+  }
+
+  @Test
+  public void testGetCdfRunJsDashboardWriteContext() {
```
Comment (davidmsantos90) [inline osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java]: We should avoid bundling multiple tests inside a unit test.

File: osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+    final CdfRunJsDashboardWriteContext cdfRunJsDashboardWriteContext = Mockito.mock( CdfRunJsDashboardWriteContext.class );
+    when( cdfRunJsDashboardWriteContext.getFactory() ).thenReturn( factory );
+    assertThat(
+      cdeEnvironment.getCdfRunJsDashboardWriteContext( cdfRunJsDashboardWriteContext, "" ),
+      instanceOf( pt.webdetails.cdf.dd.model.inst.writer.cdfrunjs.dashboard.legacy.PentahoCdfRunJsDashboardWriteContext.class ) );
+  }
+
+  @Test
+  public void testGetCdeXml() throws Exception {
```
Comment (davidmsantos90) [inline osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java]: same for this test

File: osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java (hunk 2 above)
Comment (afrjorge) [reply osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java]: its testing a NOOP dummy function required by the current core implementation, I'll remove the useless test

**Comment by afrjorge** [comment]
@graimundo @davidmsantos90 please review, applied suggested changes

**Pull request merged**